Automattic/jetpack
# Social | Initial State: Migrate URLs in the editor

Extracted out of #38983 to split it into smaller PRs.

## Proposed changes:

* Migrate URLs in the editor to the new script data

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* On a Jetpack site, login as an author, without a user connection
* Goto editor
* Open Social panel
* Confirm that the user connection link works as expected.
    <img width="289" alt="image" src="https://github.com/user-attachments/assets/2856e9f6-967b-48d1-aebb-18b31afd3b88">
* Now connect the user account.
* Now for Jetpack, Simple and Atomic sites, goto the editor
* Confirm that the connection management links/buttons work fine
* Try to break a connection, for example, if you have Facebbok or Instagram connection, remove Jetpack from [here](https://www.facebook.com/settings/?tab=business_tools) to break the connection.
* In the editor, confirm that the broken connection link points to connections manage UI for Jetpack site and to the Calypso connections management for WPCOM sites
* Likewise test that an unsupported connection link works fine, if you have one.
